sdk/core/azure-core-http-httpurlconnection/src/main/java/com/azure/core/http/httpurlconnection/SocketClient.java

```diff
@@ -0,0 +1,153 @@
+package com.azure.core.http.httpurlconnection;
+
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+
+import javax.net.ssl.SSLSocket;
+import javax.net.ssl.SSLSocketFactory;
+import java.io.*;
+import java.net.ProtocolException;
+import java.net.Socket;
+import java.net.URL;
+import java.util.*;
+
+// A socket client to be used to handle PATCH requests
+
+public class SocketClient {
+
+    private final String host;
+    private int port;
+    private final String username;
+    private final String password;
+
+    public SocketClient(String host, int port, String username, String password) {
+        this.host = host;
+        this.port = port;
+        this.username = username;
+        this.password = password;
+    }
+
+    public SocketClient(String host, int port) {
+        this(host, port, null, null);
+    }
+
+    public SocketClient(String host) {
+        this(host, 443, null, null);
+    }
+
+    public HttpUrlConnectionResponse sendPatchRequest(HttpRequest httpRequest) throws IOException {
+        if (httpRequest.getUrl().getProtocol().equals("https")) {
+            SSLSocketFactory sslSocketFactory = (SSLSocketFactory) SSLSocketFactory.getDefault();
+            try (SSLSocket socket = (SSLSocket) sslSocketFactory.createSocket(host, port)) {
+                return doInputOutput(httpRequest, socket);
+            }
+        } else if (httpRequest.getUrl().getProtocol().equals("http")) {
+            // Client is using HTTP, change port number 80
+            // Use a regular socket
+            this.port = 80;
```

Comment:
HTTP can be on any port. You should set the default port to 80 in the constructor, but don't override it here if the user has specified a different port in the constructor.